modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationParameters.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import static com.opengamma.strata.collect.Guavate.toImmutableMap;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
+
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.config.Measure;
+import com.opengamma.strata.collect.MapStream;
+
+/**
+ * The calculation parameters.
+ * <p>
+ * This provides a set of parameters that will be used in a calculation.
+ * Each parameter defines a {@linkplain CalculationParameter#queryType() query type},
+ * thus the functions are keyed in a {@code Map} by the query type {@code Class}.
+ */
+@BeanDefinition(style = "light")
+public final class CalculationParameters implements ImmutableBean, Serializable {
```

Comment:
The Javadocs don't give enough of a picture of what `CalculationParameters` are actually for.